### ISSUE TITLE
Can you provide a countdown parameter to automatic refuse the auth request?

### ISSUE THREAD

**Issue by matthew-hw** [Yubico/pam-u2f#138]
Sometimes in system GUI is unable to refuse even I unplug the device, making me have no chioce but to authorize.
It would be nice to have a countdown that can handle this disaster.
Thank you.

**Comment by a-dma** [comment]
Sorry, but I'm not sure what the problem is and what you would like to see happen.
Could you try to state that again? Maybe with a list of steps?

**Comment by daysonhw** [comment]
> Sorry, but I'm not sure what the problem is and what you would like to see happen.
> Could you try to state that again? Maybe with a list of steps?

it's a suggestion to provide a function to deny authorization, like countdown.

**Comment by jinnko** [comment]
If I'm not mistaken the ask is effectively for a timeout parameter.  Does this seem right @daysonhw?

In my case the GDM lock screen login requests a U2F touch and never seems to timeout causing the yubikey to be stuck waiting, which in turn keeps the system awake indefinitely.  Pressing ESC does seem to clear this, however if I've locked the laptop and walked away and something causes the system to wake and request auth, for example if the mouse is bumped by a passer-by, then the system just waits on the auth screen indefinitely.

Sudo authentication seems to handle this for us, however in the GDM case there seems to be no timeout.  I'm using the latest gnome-shell/wayland on a fully up-to-date ArchLinux.

Having a timeout parameter to the module would solve this for all scenarios.

For what it's worth this is my pam config saved as `/etc/pam.d/u2f+howdy`:

```
#%PAM-1.0
auth  [success=ok new_authtok_reqd=ok ignore=ignore default=1]	pam_u2f.so	authfile=/etc/u2f_mappings cue
auth  sufficient   						pam_python.so	/lib/security/howdy/pam.py
```

The effect is that if the U2F device is not available, or authentication fails (which I believe currently isn't really possible), the `default=1` will cause the howdy (face recognition) module to be skipped.

Any pam config in which I want to enable u2f+howdy then calls the above config as follows.

```
auth		include		u2f+howdy
```

**Comment by a-dma** [comment]
That sounds like some kind of issue/feature in GDM. The request will timeout if no touch is provided and pam_u2f will return an error to PAM.
